#!/usr/bin/env python

PACKAGE = "barrel_detector"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("debuging", bool_t, 0, "permision to saving converted image", False)
gen.add("probeblisitic", bool_t, 0, "permision to use HoughLinesP or HoughLine algorithm", True)
gen.add("min_radius", double_t, 0, "radius of the detected circle", 20, 0, 100)
gen.add("max_radius", double_t, 0, "radius of the detected circle", 40, 0, 100)
gen.add("min_circle_percentage", double_t, 0, "minimum percentage to accept a circle", 0.50, 0, 1)

exit(gen.generate(PACKAGE, "barrel_detector", "BarrelDetector"))
